FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1494 @@
+    #error "Please define a unique CLIENT_IDENTIFIER."
+#endif
+#ifndef BROKER_ENDPOINT
+    #error "Please define BROKER_ENDPOINT."
+#endif
+
+/**
+ * Provide default values for undefined configuration settings.
+ */
+#ifndef BROKER_PORT
+    #define BROKER_PORT    ( 1883 )
+#endif
+
+#ifndef NETWORK_BUFFER_SIZE
+    #define NETWORK_BUFFER_SIZE    ( 1024U )
+#endif
+
+/**
+ * @brief Length of client identifier.
+ */
+#define CLIENT_IDENTIFIER_LENGTH            ( ( uint16_t ) ( sizeof( CLIENT_IDENTIFIER ) - 1 ) )
+
+/**
+ * @brief Length of MQTT server host name.
+ */
+#define BROKER_ENDPOINT_LENGTH              ( ( uint16_t ) ( sizeof( BROKER_ENDPOINT ) - 1 ) )
+
+/**
+ * @brief Timeout for receiving CONNACK packet in milliseconds.
+ */
+#define CONNACK_RECV_TIMEOUT_MS             ( 1000U )
+
+/**
+ * @brief Timeout for MQTT_ProcessLoop function in milliseconds.
+ */
+#define MQTT_PROCESS_LOOP_TIMEOUT_MS        ( 200U )
+
+/**
+ * @brief The maximum time interval in seconds which is allowed to elapse
+ *  between two Control Packets.
+ *
+ *  It is the responsibility of the Client to ensure that the interval between
+ *  Control Packets being sent does not exceed the this Keep Alive value. In the
+ *  absence of sending any other Control Packets, the Client MUST send a
+ *  PINGREQ Packet.
+ */
+#define MQTT_KEEP_ALIVE_INTERVAL_SECONDS    ( 60U )
+
+/**
+ * @brief Transport timeout in milliseconds for transport send and receive.
+ */
+#define TRANSPORT_SEND_RECV_TIMEOUT_MS      ( 20 )
+
+#define _MILLISECONDS_PER_SECOND            ( 1000U )                                                 /**< @brief Milliseconds per second. */
+#define _MILLISECONDS_PER_TICK              ( _MILLISECONDS_PER_SECOND / configTICK_RATE_HZ )         /**< Milliseconds per FreeRTOS tick. */
+
+/**
+ * @brief Ticks to wait for task notifications.
+ */
+#define DEMO_TICKS_TO_WAIT                  pdMS_TO_TICKS( 1000 )
+
+/**
+ * @brief Maximum number of operations awaiting an ack packet from the broker.
+ */
+#define PENDING_ACKS_MAX_SIZE               20
+
+/**
+ * @brief Maximum number of subscriptions to store in the subscription list.
+ */
+#define SUBSCRIPTIONS_MAX_COUNT             10
+
+/**
+ * @brief Number of publishes done by the publisher in this demo.
+ */
+#define PUBLISH_COUNT                       16
+
+/**
+ * @brief Size of statically allocated buffers for holding topic names and payloads in this demo.
+ */
+#define DEMO_BUFFER_SIZE                    100
+
+/**
+ * @brief Size of dynamically allocated buffers for holding topic names and payloads in this demo.
+ */
+#define DYNAMIC_BUFFER_SIZE                 25
+
+/**
+ * @brief Max number of commands that can be enqueued.
+ */
+#define COMMAND_QUEUE_SIZE                  25
+
+/**
+ * @brief Max number of received publishes that can be enqueued for a task.
+ */
+#define PUBLISH_QUEUE_SIZE                  20
+
+/**
+ * @brief Delay for the subscriber task when no publishes are waiting in the queue.
```

Comment:
I couldn't understand how the subscriber task will use this delay. A bit more clarity in this description will be good.